samples/nrf9160/modem_trace_flash/src/main.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/uart.h>
+#include <modem/lte_lc.h>
+#include <modem/nrf_modem_lib.h>
+#include <modem/nrf_modem_lib_trace.h>
+#include <nrf_modem.h>
+#include <nrf_modem_at.h>
+#include <dk_buttons_and_leds.h>
+
+#include "trace_flash.h"
+
+LOG_MODULE_REGISTER(modem_trace_flash_sample, CONFIG_MODEM_TRACE_FLASH_SAMPLE_LOG_LEVEL);
+
+#define UART1_DT_NODE DT_NODELABEL(uart1)
+
+static const struct device *const uart_dev = DEVICE_DT_GET(UART1_DT_NODE);
+
+/* define callback */
+LTE_LC_ON_CFUN(cfun_hook, on_cfun, NULL);
+
+/* callback implementation */
+static void on_cfun(enum lte_lc_func_mode mode, void *context)
+{
+	LOG_INF("LTE mode changed to %d\n", mode);
+}
+
+static void print_uart1(char *buf, int len)
+{
+	if (!device_is_ready(uart_dev)) {
+		LOG_ERR("uart1 device not found!");
+		return;
+	}
+	for (int i = 0; i < len; i++) {
+		uart_poll_out(uart_dev, buf[i]);
+	}
+}
+
+static void print_traces(void)
+{
+	uint8_t buf[1024];
```

Comment:
```suggestion
        uint8_t buf[READ_BUF_SIZE];
```
And define READ_BUF_SIZE to 1024 at the top of the file.

Reply:
Why not just use a `const` inside the function? It's only needed here.

Reply:
It is better to clearly explain why a magic number was chosen. A macro in place of a magic number becomes a self-explanatory way of doing that.

Reply:
Yes, I agree to avoid the magic number, but I don't see the reason to use a define that would be file global instead of a const that would only be visible inside the function scope.

Reply:
oh yeah! sorry I misunderstood. A const works too! :)